### PR TITLE
tandoor-recipes: 1.5.12 -> 1.5.16

### PR DESCRIPTION
**Pull request by sinavir**
## Description of changes

https://github.com/TandoorRecipes/recipes/releases/tag/1.5.16

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ambroisie** [comment]
Mmh, why wasn't it updated by r-ryantm?

File: pkgs/applications/misc/tandoor-recipes/default.nix
Comment (ambroisie) [inline pkgs/applications/misc/tandoor-recipes/default.nix:133]: Please sort the list.

I also wonder whether `pytest-xdist` should be used on Hydra.

**Review by ambroisie — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/tandoor-recipes/default.nix
Comment (sinavir) [reply pkgs/applications/misc/tandoor-recipes/default.nix]: I don't know. I just checked the requirements.txt diff and didn't thought about it.

I'm am currently checking if it builds without

Comment (sinavir) [reply pkgs/applications/misc/tandoor-recipes/default.nix]: We need to apply this patch if we want to get rid of `pytest-xdist` :
```
diff --git a/pytest.ini b/pytest.ini
index a3d26ec4..8bdf12fb 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -3,5 +3,5 @@ DJANGO_SETTINGS_MODULE = recipes.test_settings
 testpaths = cookbook/tests
 python_files = tests.py test_*.py *_tests.py
 # uncomment to run coverage reports
-addopts = -n auto --cov=. --cov-report=html:docs/reports/coverage --cov-report=xml:docs/reports/coverage/coverage.xml --junitxml=docs/reports/tests/pytest.xml --html=docs/reports/tests/tests.html
+addopts = --cov=. --cov-report=html:docs/reports/coverage --cov-report=xml:docs/reports/coverage/coverage.xml --junitxml=docs/reports/tests/pytest.xml --html=docs/reports/tests/tests.html
 # addopts = -n auto --junitxml=docs/reports/tests/pytest.xml --html=docs/reports/tests/tests.html
\ No newline at end of file
```

I'm not aware of hydra limitations about parallelism. But my guess is that we have those two solutions regarding to how hydra works :
 - The sandbox hides the "unused" cpus so we need to change nothing
 - We keep `pytest-xdist` but change the `-n auto` to `-n $NIX_BUILD_CORES` (expanding the variable this I don't think `pytest.ini` support variable expansion)

Comment (sinavir) [reply pkgs/applications/misc/tandoor-recipes/default.nix]: Turns out we should use the second option according to what I read. I pushed an implementation accordingly

File: pkgs/applications/misc/tandoor-recipes/default.nix
Comment (ambroisie) [inline pkgs/applications/misc/tandoor-recipes/default.nix:141]:
```suggestion
    pytestCheckHook
    mock
    pytest-asyncio
    pytest-cov
    pytest-django
    pytest-factoryboy
    pytest-html
    pytest-xdist
```

**Review by ambroisie — CHANGES_REQUESTED** [review]
Will add more knowledgeable python reviewers to approve the `pytest-xdist` bits.

**Comment by ambroisie** [comment]
@FRidh, @tjni, @jonringer could (one of) you take a look at the `pytest-xdist` bits please? Thank you :-)

**Review by jonringer — COMMENTED** [review]
seems like this might have a data race:
```
________________________ ERROR at setup of test_delete ________________________
...
self = <django.db.backends.sqlite3.base.SQLiteCursorWrapper object at 0x7fffe67b2960>
query = 'INSERT INTO "auth_user" ("password", "last_login", "is_superuser", "username", "first_name", "last_name", "email", ">
params = ('', None, False, 'lori35', 'James', 'Martin', ...)

    def execute(self, query, params=None):
        if params is None:
            return super().execute(query)
        # Extract names if params is a mapping, i.e. "pyformat" style is used.
        param_names = list(params) if isinstance(params, Mapping) else None
        query = self.convert_query(query, param_names=param_names)
>       return super().execute(query, params)
E       django.db.utils.IntegrityError: UNIQUE constraint failed: auth_user.username
```

**Comment by jonringer** [comment]
maybe add `test_delete` to flakey tests?

**Comment by sinavir** [comment]
Will do. It never failed on my comptuer.

**Review by jonringer — APPROVED** [review]
LGTM

```
Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/302129

1 package built:
tandoor-recipes
```

**Comment by jonringer** [comment]
blocked until @ambroisie approves :)

**Comment by ambroisie** [comment]
Thanks for the review @jonringer.

Everything looks good to me, merging :shipit:.

**Review by ambroisie — APPROVED** [review]

**Pull request merged**
